scrapegraphai/nodes/rag_node.py

```diff
@@ -98,7 +99,19 @@ def execute(self, state: dict) -> dict:
         )
         embeddings = self.embedder_model
 
-        retriever = FAISS.from_documents(chunked_docs, embeddings).as_retriever()
+        folder_name = "cache"
```

Comment:
that's the only thing I'd change.

as it stands, each RAG node with the cache config would overwrite the very same folder, effectively nullifying the cache, unless a RAG node on the very same input state is invoked subsequently.

I would allora more flexibility by either 1): allowing the user to specify the cache folder in the config for that specific RAG node or 2) automatically compute a unique identifier for that specific input state (e.g., the filename, the endpoint or a some hashing function on the documents) so that the user may choose what to keep and what not to keep in the cache